pkg/app/piped/sourceprocesser/sourceprocesser.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2024 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sourceprocesser
+
+import (
+	"fmt"
+)
+
+type SourceTemplateProcessor interface {
+	// BuildTemplateData returns the data that will be used to template target files
+	BuildTemplateData(appDir string) (map[string]string, error)
+	// TemplateSource performs the templating prepared data to the source files
```

Comment:
You mean this?
```suggestion
	// TemplateSource applies prepared template data to the source files
```

Reply:
Thanks, addressed by https://github.com/pipe-cd/pipecd/pull/4855/commits/25acbb46d0b44cd51d48a6d659356d91ac98a528